plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/reporter/Reporter.kt

```diff
@@ -0,0 +1,59 @@
+// Collection of functions that applies output Ktlint reporter
+
+package org.jlleitschuh.gradle.ktlint.reporter
+
+import org.gradle.api.Project
+import org.gradle.api.tasks.JavaExec
+import org.gradle.util.GFileUtils
+import java.io.File
+import net.swiftzer.semver.SemVer
+import org.jlleitschuh.gradle.ktlint.KtlintExtension
+import java.io.FileOutputStream
+import java.io.OutputStream
+
+/**
+ * Apply reporter to the task.
+ */
+internal fun JavaExec.applyReporters(target: Project, extension: KtlintExtension, sourceSetName: String) {
+    if (SemVer.parse(extension.version).compareTo(SemVer(0, 10, 0)) >= 0) {
+        extension.reporters.forEach { reporter ->
+            doFirst {
+                val reportOutputFile = createReportOutputFile(target, reporter.fileExtension, sourceSetName)
+                this.args("--reporter=${reporter.reporterName},output=${reportOutputFile.absolutePath}")
+            }
+        }
+        if (extension.outputToConsole) {
+            this.args("--reporter=plain")
+        }
+    } else {
+        extension.reporters.firstOrNull()?.let { reporter ->
+            if (SemVer.parse(extension.version).compareTo(reporter.availableSinceVersion) >= 0) {
+                var reportOutput: FileOutputStream? = null
+                doFirst {
+                    reportOutput = createReportOutputFile(target, reporter.fileExtension, sourceSetName).outputStream().also {
+                        this.args("--reporter=${reporter.reporterName}")
+                        if (extension.outputToConsole) {
+                            this.standardOutput = object : OutputStream() {
+                                override fun write(b: Int) {
+                                    it.write(b)
+                                    System.out.write(b)
```

Comment:
Was this left in intentionally??

Reply:
Yes, if the `outputToConsole` is `true`, put all output to console as well.

Reply:
@Tapchicoma Should this be printed to a logger or to `System.out`?

Reply:
Ktlint with version 0.10.0 and lower does not support multiple reporters. So if the developer wants to `outputToConsole` and reporting to file in same time, I need to transfer output from ktlint to console and file in one go.

Reply:
Right, that makes sense.